pkg/cmd/server/origin/master_config.go

```diff
@@ -252,28 +253,26 @@ func newAuthenticator(config configapi.MasterConfig, etcdHelper tools.EtcdHelper
 	return ret
 }
 
-func newProjectAuthorizationCache(openshiftClient *osclient.Client, kubeClient *kclient.Client) *projectauth.AuthorizationCache {
+func newProjectAuthorizationCache(openshiftClient *osclient.Client, kubeClient *kclient.Client,
+	policyClient policyclient.ReadOnlyPolicyClient) *projectauth.AuthorizationCache {
 	return projectauth.NewAuthorizationCache(
 		projectauth.NewReviewer(openshiftClient),
 		kubeClient.Namespaces(),
-		openshiftClient,
-		openshiftClient,
-		openshiftClient,
-		openshiftClient,
+		policyClient,
 	)
 }
 
-func newPolicyCache(etcdHelper tools.EtcdHelper) *policycache.PolicyCache {
+func newReadOnlyCache(etcdHelper tools.EtcdHelper) *policycache.ReadOnlyCache {
```

Comment:
it returns a policycache.ReadOnlyCache

what happens when we have other shared caches in future?

i.e. *podcache.ReadOnlyCache, *namespacecache.ReadOnlyCache, etc.